instrumentation/pg/lib/opentelemetry/instrumentation/pg/patches/connection.rb

```diff
@@ -15,6 +15,8 @@ module Patches
         module Connection # rubocop:disable Metrics/ModuleLength
           PG::Constants::EXEC_ISH_METHODS.each do |method|
             define_method method do |*args|
+              return super(*args) unless config[:trace_root_spans] || OpenTelemetry::Trace.current_span.context.valid?
```

Comment:
```suggestion
              return super(*args) unless config[:trace_root_spans] || OpenTelemetry::Trace.current_span.recording?
```

Reply:
does this change do the same? I think it's a little different

Reply:
It does the same thing. This is the officially supported way to decide whether to do "expensive" things, based on whether the data will actually be recorded.